tests/misc_test.go

```diff
@@ -283,3 +284,21 @@ func zero() int {
 	}()
 	panic("")
 }
+
+func TestNumGoroutine(t *testing.T) {
+	n := runtime.NumGoroutine()
+	c := make(chan bool)
+	go func() {
+		<-c
+		<-c
+		<-c
+		<-c
+	}()
+	c <- true
+	c <- true
+	c <- true
+	if g, w := runtime.NumGoroutine(), n+1; g != w {
```

Comment:
I think it'd be nicer to fully spell out the variables `g`, `w` as `got` and `want`. That seems to be more common, and IMO, more clear.

The rest of this test looks good to me.